rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1032,78 +1032,133 @@ private void processReleaseRequests() {
+                    // Unexpected error accessing storage, log and fail
+                    logger.error("Unexpected error accessing storage while attempting to add a release BTC tx", exception);
+                    return;
+                }
 
-            List<UTXO> selectedUTXOs = result.get().getSelectedUTXOs();
-            BtcTransaction generatedTransaction = result.get().getBtcTx();
-            List<UTXO> availableUTXOs;
-            ReleaseTransactionSet releaseTransactionSet;
+                BtcTransaction generatedTransaction = result.get().getBtcTx();
+                addBtcTxToReleaseTransactionSet(generatedTransaction, releaseTransactionSet, null);
+
+                // Mark UTXOs as spent
+                List<UTXO> selectedUTXOs = result.get().getSelectedUTXOs();
+                availableUTXOs.removeAll(selectedUTXOs);
+
+                // update next Pegout height
+                long nextPegoutHeight = currentBlockNumber + bridgeConstants.getNumberOfBlocksBetweenPegouts();
+                provider.setNextPegoutHeight(nextPegoutHeight);
+                provider.saveNextPegoutHeight();
+
+                // TODO: Update Adjust balances in edge cases
 
-            // Attempt access to storage first
-            // (any of these could fail and would invalidate both
-            // the tx build and utxo selection, so treat as atomic)
-            try {
-                availableUTXOs = getActiveFederationBtcUTXOs();
-                releaseTransactionSet = provider.getReleaseTransactionSet();
-            } catch (IOException exception) {
-                // Unexpected error accessing storage, log and fail
-                logger.error(
-                        String.format(
-                                "Unexpected error accessing storage while attempting to add a release BTC tx for <%s, %s>",
-                                releaseRequest.getDestination().toString(),
-                                releaseRequest.getAmount().toString()
-                        ),
-                        exception
-                );
-                return false;
             }
 
-            if (activations.isActive(ConsensusRule.RSKIP146)) {
-                Keccak256 rskTxHash = releaseRequest.getRskTxHash();
-                // Add the TX
-                releaseTransactionSet.add(generatedTransaction, rskExecutionBlock.getNumber(), rskTxHash);
-                // For a short time period, there could be items in the release request queue that don't have the rskTxHash
-                // (these are releases created right before the consensus rule activation, that weren't processed before its activation)
-                // We shouldn't generate the event for those releases
-                if (rskTxHash != null) {
-                    // Log the Release request
-                    eventLogger.logReleaseBtcRequested(rskTxHash.getBytes(), generatedTransaction, releaseRequest.getAmount());
+        } else {
```

Comment:
Let's move pre and post fork logic to their own methods. `processReleasesIndividually` and `processReleasesInBatch`

Reply:
Done